res/qml/LibraryControl.qml

```diff
@@ -0,0 +1,187 @@
+import Mixxx 0.1 as Mixxx
+import QtQuick 2.12
+
+Item {
```

Comment:
whats the purpose of this component exactly?

Reply:
It's the QML version of the current `LibraryControl` defined in `librarycontrol.cpp`. As all EngineControl classes, these belong into the backend and handle the interaction with the control objects. The current `LibraryControl` is unfortunately a big mess, because it directly interacts with the QWidgets. This is why it basically needs to be rewritten for QML.

Ideally, the `LibraryControl` would just manage the COs and then emit signals like "load selected track", "move selection down", "set focus to track table", etc. and then the GUI code listens for those signals and acts based on them.

Reply:
Mhmm ok. I'm a bit weirded out by the fact that we use signals for requesting actions to occur somewhere else. afaik that is not very idiomatic Qt.

Reply:
> Mhmm ok. I'm a bit weirded out by the fact that we use signals for requesting actions to occur somewhere else. afaik that is not very idiomatic Qt.

Well, what I tried to avoid is to reimplement the LibraryControl functionality in each QML skin individually. To make the LibraryControl component reusable, it does not make any assumptions about the library model/view used by skin, and still requires only a minimum amount of code. The skin designer just needs to hook up the signals to the library view.

If you have better idea how to archieve separation of concerns and resuability, I'm all ears. 🙂

Reply:
Yeah. Not really either, using signals to achieve weak coupling seems like a good idea but odd at the same time.

Reply:
We could even move all the functionality of `LibraryControl.qml` into the C++ LibraryControl. Maybe that would make it a bit less weird? I didn't do that for now, because then the `librarycontrol.cpp` code would become a complete mess since it still needs to support the old QWidgets UI.

Reply:
We might want to consider that in the future sure. Don't get me wrong, I consider QML LibraryControl a more than decent solution currently.